### PR TITLE
gh-121533: Improve `PyCell_[Get,Set]` docs: mention the exceptions

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-121533

----
📚 Documentation preview 📚: https://cpython-previews--121534.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/cell.rst [hunk 1]
```diff
@@ -39,7 +39,8 @@ Cell objects are not likely to be useful elsewhere.
 
 .. c:function:: PyObject* PyCell_Get(PyObject *cell)
 
-   Return the contents of the cell *cell*.
+   Return the contents of the cell *cell*. If *cell* is not a cell object,
```
Comment (vstinner) [inline Doc/c-api/cell.rst]: You should explain that the cell contents **can** be NULL.

File: Doc/c-api/cell.rst [hunk 2]
```diff
@@ -52,8 +53,8 @@ Cell objects are not likely to be useful elsewhere.
 
    Set the contents of the cell object *cell* to *value*.  This releases the
    reference to any current content of the cell. *value* may be ``NULL``.  *cell*
-   must be non-``NULL``; if it is not a cell object, ``-1`` will be returned.  On
-   success, ``0`` will be returned.
+   must be non-``NULL``; if it is not a cell object, ``-1`` will be returned
+   with an exception set.  On success, ``0`` will be returned.
```
Comment (vstinner) [inline Doc/c-api/cell.rst]:
```suggestion
   must be non-``NULL``.
   
   On success, return ``0``.
   If *cell* is not a cell object, set an exception and return ``-1``.
```

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
The Lint job failed with a space issue.

**Comment by sobolevn** [comment]
@vstinner thanks a lot for the quick review!

**Comment by sobolevn** [comment]
I created two backports, because these functions are present on 3.12 and 3.13 and behave the same way.

**Pull request merged**